code/modules/multiz/movement.dm

```diff
@@ -0,0 +1,342 @@
+		return 1
+
+/* Same as above, hull scaling discussion pending
+
+	for(var/turf/simulated/T in trange(1,src)) //Robots get "magboots"
+		if(T.density)
+			return 1*/
+
+// TODO - Leshana Experimental
+
+//Execution by grand piano!
+/atom/movable/proc/get_fall_damage()
+	return 42
+
+//If atom stands under open space, it can prevent fall, or not
+/atom/proc/can_prevent_fall(var/atom/movable/mover, var/turf/coming_from)
+	return (!Cross(mover, coming_from))
+
+/atom/proc/get_gravity()
+	var/area/A = get_area(src)
+	if(istype(A))
+		return A.gravity
+
+	return 1
+
+////////////////////////////
+
+//FALLING STUFF
+/atom
+	var/fall_lock = FALSE // Stops fall() being called during gravity spawn delay
+	var/zs_fallen = 0 // Gets reset if it hits something, for fall damage
+
+//Holds fall checks that should not be overriden by children
+/atom/movable/proc/fall()
+	if(fall_lock)
+		return
+
+	if(!isturf(loc))
+		return
+
+	var/turf/below = GetBelow(src)
+	if(!below)
+		return
+
+	var/turf/bottom = null
+	for(bottom = GetBelow(src); isopenspace(bottom); bottom = GetBelow(bottom))
+
+	if(istype(bottom,/turf/space))
+		return
+
+	var/turf/T = loc
+	if(!T.CanZPass(src, DOWN) || !below.CanZPass(src, DOWN))
+		return
+
+	var/gravity = get_gravity()
+	// No gravity in space, apparently.
+	if(!gravity) //Polaris uses a proc, has_gravity(), for this
+		return
+	fall_lock = TRUE
+	spawn(4 / gravity) // Now we use a delay of 4 ticks divided by the gravity.
+		fall_lock = FALSE
+		
+		// We're in a new loc most likely, so check all this again
+		below = GetBelow(src)
+		if(!below)
+			return
+	
+		bottom = null
+		for(bottom = GetBelow(src); isopenspace(bottom); bottom = GetBelow(bottom))
+	
+		if(istype(bottom,/turf/space))
+			return
+		T = loc
+		if(!T.CanZPass(src, DOWN) || !below.CanZPass(src, DOWN))
+			return
+	
+		gravity = get_gravity()
+		if(!gravity)
+			return
+
+		/*if(throwing)  This was causing odd behavior where things wouldn't stop.
+			return*/
+	
+		if(can_fall())
+			// We spawn here to let the current move operation complete before we start falling. fall() is normally called from
+			// Entered() which is part of Move(), by spawn()ing we let that complete.  But we want to preserve if we were in client movement
+			// or normal movement so other move behavior can continue.
+			var/mob/M = src
+			var/is_client_moving = (ismob(M) && M.client && M.client.moving)
+			spawn(0)
+				if(is_client_moving) M.client.moving = 1
+				handle_fall(below)
+				if(is_client_moving) M.client.moving = 0
+			// TODO - handle fall on damage!
+
+//For children to override
+/atom/movable/proc/can_fall()
+	if(anchored)
+		return FALSE
+	return TRUE
+
+/obj/effect/can_fall()
+	return FALSE
+
+/obj/effect/decal/cleanable/can_fall()
+	return TRUE
+
+// These didn't fall anyways but better to nip this now just incase.
+/atom/movable/lighting_overlay/can_fall()
+	return FALSE
+
+// Mechas are anchored, so we need to override.
+/obj/mecha/can_fall()
```

Comment:
as the Clarke has a built in 'jetpack', this should probably be false for it

Reply:
Fixed, also misread as move to new file but it's fine